src/main/java/com/es/phoneshop/model/product/Product.java

```diff
@@ -92,4 +93,19 @@ public String getImageUrl() {
     public void setImageUrl(String imageUrl) {
         this.imageUrl = imageUrl;
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
check on id is enough, because only this field is expected to be unique and we can rely on this fact